doc/user/content/transform-data/patterns/time-travel-queries.md

```diff
@@ -0,0 +1,153 @@
+
+The history retention period represents the minimum amount of historical data
+guaranteed to be retained by Materialize. History clean-up is processed in the
+background, so older history may be accessible for the period of time between
+when it falls outside the retention period and when it is cleaned up.
+
+### Best practices
+
+Given the additional memory costs associated with increasing the history
+retention period on indexes, if you don't need the performance benefits of an
+index - in particular for the use case of [durable subscriptions](/sql/subscribe#durable-subscriptions) -
+you should consider creating a materialized view for your subscription query
+and configure the history retention period on that materialized view
+instead.
+
+Similarly, because of the increased storage costs and processing time for the
+additional historical data on whichever object has an increased history
+retention period, you should consider configuring history retention period on
+the index or materialized view directly powering the subscription, rather than
+all the way through the dependency chain from the source to the index or
+materialized view.
+
+## Examples
+
+### Enabling history retention
+
+To set a history retention period for an object, use the `RETAIN HISTORY`
+option, which accepts positive [interval](/sql/types/interval/) values
+(e.g. `'1hr'`):
+
+```sql
+CREATE MATERIALIZED VIEW winning_bids
+WITH (RETAIN HISTORY FOR '1hr') AS
+SELECT auction_id,
+       bid_id,
+       item,
+       amount
+FROM highest_bid_per_auction
+WHERE end_time < mz_now();
+```
+
+To adjust the history retention period for an object, use `ALTER`:
+
+```sql
+ALTER MATERIALIZED VIEW winning_bids SET (RETAIN HISTORY FOR '2hr');
+```
+
+<!-- TODO(maddyblue): replace this paragraph with a mention of the catalog
+     table/column once it's available -->
+To see what history retention period has been configured for an object, run the
+`SHOW CREATE ...` statement for the given object. See [`SHOW CREATE SOURCE`](/sql/show-create-source/)
+for an example.
+
+```sql
+SHOW CREATE MATERIALIZED VIEW winning_bids;
+```
+```nofmt
+              name               |                                                                                                                       create_sql
+---------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+ materialize.public.winning_bids | CREATE MATERIALIZED VIEW "materialize"."public"."winning_bids" IN CLUSTER "quickstart" WITH (RETAIN HISTORY = FOR '2hr') AS SELECT * FROM "materialize"."public"."highest_bid_per_auction" WHERE "end_time" < "mz_catalog"."mz_now"()
+```
+
+### Disabling history retention
+
+To disable history retention, set the `RETAIN HISTORY` option to its original
+default value (1 second):
```

Comment:
```suggestion
To disable history retention, reset the `RETAIN HISTORY` option:
```

Reply:
Agree to your suggestion - one wrinkle is that `mz_internal.mz_history_retention_strategies` returns 
```
strategy |  value  
---------+---------
FOR      | 1000
```
when retain history is unset. Is it confusing to say everywhere "no retain history", but then this table doesn't say zero it says 1 second? Should we put a small note somewhere that the default/"no retain history" == 1 second

Reply:
I had similar concerns when writing this, and I think that we do want to expose exactly what the current retain history is in ms. We could document this somewhere. We also could add a third column `is_default` or something that's a bool? I'm fine with any solution here.